Sources/XRPLSwift-rest/Utils/Http.swift

```diff
@@ -0,0 +1,105 @@
+//
+//  http.swift
+//
+//
+//  Created by Bryan on 7/12/24.
+//
+
+import Foundation
+import AnyCodable
+
+typealias APIParams = [String: Any?]
+
+let TimeoutInterval = TimeInterval(3)
+
+open class BaseResponse<T: Codable>: Codable {
+    public var result: T
+    
+    private enum CodingKeys: String, CodingKey {
+        case result = "result"
+    }
+}
+
+open class ErrorResponse: Codable {
+    public var status: String = "error"
+    public var error: String
+    public var errorCode: Int?
+    public var errorMessage: String?
+    public var request: AnyCodable?
+
+    private enum CodingKeys: String, CodingKey {
+        case status = "status"
+        case error = "error"
+        case errorCode = "error_code"
+        case errorMessage = "error_message"
+        case request = "request"
+    }
+}
+
+class http {
+    enum APIError : Error, Equatable {
+        case invalidURL
+        case noData
+        case httpResponseIsNil
+        case ErrorResponse(code: Int?, msg: String?)
+        case unknownError
+    }
+
+    enum HttpMethod: String {
+        case put = "PUT"
+        case post = "POST"
+        case get = "GET"
+        case delete = "DELETE"
+    }
+
+    static func request<T: Codable>(httpMethod : HttpMethod, url: URL, params: APIParams? = nil) async throws -> T? {
+        let components = URLComponents(url: url, resolvingAgainstBaseURL: true)
+        
+        guard let finalURL = components?.url else {
+            throw APIError.invalidURL
+        }
+        
+        var request = URLRequest(url: finalURL)
+        request.httpMethod = httpMethod.rawValue
+        request.timeoutInterval = TimeoutInterval
+        
+        request.addValue("application/json", forHTTPHeaderField: "Content-Type")
+        
+        if params != nil && ![.get, .delete].contains(httpMethod) {
+            do {
+                request.httpBody = try JSONSerialization.data(withJSONObject: params as Any)
+            } catch {
+                print("Error creating JSON data: \(error)")
+            }
+        }
```

Comment:
How about making the interval for timeout optional as well, and organizing optional values into a dedicated struct for options? I think this would make it easier to work with options later on.

```swift
public struct RequestOptions {
    let params: APIParams?
    let timeoutInterval: TimeInterval
    
    init(params: APIParams? = nil, timeoutInterval: TimeInterval = TimeoutInterval) {
        self.params = params
        self.timeoutInterval = timeoutInterval
    }
}
```

```suggestion
    static func request<T: Codable>(httpMethod : HttpMethod, url: URL, options: RequestOptions = RequestOptions()) async throws -> T? {
        let components = URLComponents(url: url, resolvingAgainstBaseURL: true)
        
        guard let finalURL = components?.url else {
            throw APIError.invalidURL
        }
        
        var request = URLRequest(url: finalURL)
        request.httpMethod = httpMethod.rawValue
        request.timeoutInterval = options.timeoutInterval
        
        request.addValue("application/json", forHTTPHeaderField: "Content-Type")
        
        if let params = options.params, ![.get, .delete].contains(httpMethod) {
            do {
                request.httpBody = try JSONSerialization.data(withJSONObject: params)
            } catch {
                print("Error creating JSON data: \(error)")
            }
        }
```

Reply:
If creating a struct called `Options` doesn't seem like a good idea, how about just applying [Optional Binding](https://developer.apple.com/documentation/swift/optional#Optional-Binding)?
That way, you might not need to do any type casting.

```suggestion
    static func request<T: Codable>(httpMethod : HttpMethod, url: URL, params: APIParams? = nil) async throws -> T? {
        let components = URLComponents(url: url, resolvingAgainstBaseURL: true)
        
        guard let finalURL = components?.url else {
            throw APIError.invalidURL
        }
        
        var request = URLRequest(url: finalURL)
        request.httpMethod = httpMethod.rawValue
        request.timeoutInterval = TimeoutInterval
        
        request.addValue("application/json", forHTTPHeaderField: "Content-Type")
        
        if let params = params, ![.get, .delete].contains(httpMethod) {
            do {
                request.httpBody = try JSONSerialization.data(withJSONObject: params)
            } catch {
                print("Error creating JSON data: \(error)")
            }
        }
```

Reply:
Adopted the latter one

[f60818b](https://github.com/girin-app/XRPLSwift-rest/pull/1/commits/f60818b9ac845fd3c9867d4babbeb1d5c883d815)